pkg/client/generic_client.go

```diff
@@ -0,0 +1,263 @@
+			return errors.New(errors.ClientError, "It looks like the transport protocol does not support transmitting biz error, use TTHeader please")
+		}
+		opts = append(opts, client.WithTransportProtocol(transport.TTHeader))
+		opts = append(opts, client.WithMetaHandler(transmeta.ClientTTHeaderHandler))
+	} else {
+		if c.Conf.Transport != "" {
+			switch c.Conf.Transport {
+			case "TTHeader":
+				opts = append(opts, client.WithTransportProtocol(transport.TTHeader))
+			case "Framed":
+				opts = append(opts, client.WithTransportProtocol(transport.Framed))
+			case "TTHeaderFramed":
+				opts = append(opts, client.WithTransportProtocol(transport.TTHeaderFramed))
+			}
+		}
+	}
+
+	c.ClientOpts = opts
+	return nil
+}
+
+func (c *GenericClientBase) BuildCallOptions() (context.Context, error) {
+	ctx := context.Background()
+	var opts []callopt.Option
+
+	// Add metainfo to context
+	if c.Conf.Transport != "TTHeader" &&
+		(len(c.Conf.Meta) != 0 || len(c.Conf.MetaPersistent) != 0 || c.Conf.MetaBackward) {
+		return nil, errors.New(errors.ClientError, "It looks like the protocol does not support transmitting meta information")
+	}
+
+	if c.Conf.Transport == "TTHeader" {
+		if c.Conf.Meta != nil {
+			for k, v := range c.Conf.Meta {
+				ctx = metainfo.WithValue(ctx, k, v)
+			}
+		}
+		if c.Conf.MetaPersistent != nil {
+			for k, v := range c.Conf.MetaPersistent {
+				ctx = metainfo.WithPersistentValue(ctx, k, v)
+			}
+		}
+		if c.Conf.MetaBackward {
+			// must mark the context to receive backward meta information
+			ctx = metainfo.WithBackwardValues(ctx)
+		}
+	}
+	c.CallOptions = opts
+	return ctx, nil
+}
+
+func (c *GenericClientBase) HandleBizError(bizErr kerrors.BizStatusErrorIface) error {
+	var r string
+	var err error
+	if len(bizErr.BizExtra()) != 0 {
+		r, err = log.FormatMap(bizErr.BizExtra())
+		if err != nil {
+			return err
+		}
+	}
+	log.Success()
+	log.Println("\033[32mReceived metainfo from server: \033[0m")
+	log.Printf("BizStatusCode: %d\n", bizErr.BizStatusCode())
+	log.Printf("BizMessage: %s\n", bizErr.BizMessage())
+
+	if len(bizErr.BizExtra()) != 0 {
+		log.Println("BizExtra:")
+		log.Println(r)
+	}
+	return nil
+}
+
+func (c *GenericClientBase) GetResponse() interface{} {
+	return c.Resp
+}
+
+func (c *GenericClientBase) GetMetaBackward() map[string]string {
+	return c.MetaBackward
+}
+
+type ThriftGeneric struct {
+	GenericClientBase
+	Provider generic.DescriptorProvider
+}
+
+func NewThriftGeneric() *ThriftGeneric {
+	return &ThriftGeneric{}
+}
+
+func (c *ThriftGeneric) Init(Conf *config.Config) error {
+	// Waiting for server reflection
+	p, err := generic.NewThriftFileProvider(Conf.IDLPath)
+	if err != nil {
+		return err
+	}
+	c.Provider, c.Conf = p, Conf
+
+	g, err := generic.JSONThriftGeneric(p)
+	if err != nil {
+		return err
+	}
+	c.Generic = g
+
+	if err := c.BuildClientOptions(); err != nil {
+		return err
+	}
+
+	cli, err := genericclient.NewClient(Conf.IDLServiceName, c.Generic, c.ClientOpts...)
```

Comment:
`IDLServiceName` should not used as the `DiscoveryServiceName`. Though we do need a placeholder here and it can be a global constant before we introduce the discovery implementations. And `IDLServiceName` (if specified by user) should be set to TTHeader (if transport is TTHeader/TTHeaderFramed) as the value of the strKey "isn"

Reply:
Here's an example of writing the "isn" to TTHeader:

```go
import (
    "github.com/cloudwego/kitex/pkg/remote"
    "github.com/cloudwego/kitex/pkg/remote/transmeta"
)

idlServiceName := "<IDL Service Name>"

opts := []client.Option{
    client.WithTransportProtocol(transport.TTHeader), // necessary for multiservice
    client.WithMetaHandler(remote.NewCustomMetaHandler(
       remote.WithWriteMeta(func(ctx context.Context, msg remote.Message) (context.Context, error) {
          strInfo := msg.TransInfo().TransStrInfo()
          strInfo[transmeta.HeaderIDLServiceName] = idlServiceName
          return ctx, nil
       }),
    )),
}
cli, err := genericclient.NewClient("your server", g, opts...)
```

Reply:
done